src/Resources/views/CRUD/base_array_macro.html.twig

```diff
@@ -8,17 +8,75 @@ For the full copyright and license information, please view the LICENSE
 file that was distributed with this source code.
 
 #}
-{% macro render_array(value, inline) %}
-    {% from _self import render_array %}
-    {% for key, val in value %}
-        {% if val is iterable %}
-            [{{ key }} => {{ render_array(val, inline) }}]
+{# NEXT_MAJOR: move inline parameter under options #}
+{% macro render_array(value, inline, options = []) %}
+    {% spaceless %}
+        {% from _self import render_array %}
+
+        {% if not inline %}
+            <ul>
         {% else %}
-            [{{ key }} => {{ val }}]
+            [
         {% endif %}
 
-        {% if not loop.last and not inline %}
-            <br>
+        {% for key, val in value %}
+            {% if val is not iterable and not inline %}
+                <li>
+            {% endif %}
+
+            {% if options.display in ['both', 'keys'] or options.display is not defined %}
+                {% if options.key_translation_domain is same as(false) or options.key_translation_domain is not defined %}
+                    {{ key }}
+                {% else %}
+                    {% if options.key_translation_domain is same as(true) %}
```

Comment:
What happen if key_translation_domain is not defined ?

Reply:
On the 28 line we check this condition with `options.key_translation_domain is not defined` (will set this check first)

Reply:
Oh ! I missed it, sorry.